samcli/lib/utils/git_repo.py

```diff
@@ -118,7 +119,7 @@ def clone(self, clone_dir: Path, clone_name: str, replace_existing: bool = False
                 temp_path = os.path.normpath(os.path.join(tempdir, clone_name))
                 git_executable: str = GitRepo._git_executable()
                 LOG.info("\nCloning from %s", self.url)
-                subprocess.check_output(
+                check_output(
                     [git_executable, "clone", self.url, clone_name],
```

Comment:
What if we shallow-clone instead? So that everyone benefits from `sam init` and there's no risk to the patched test code diverging from prod code.

Reply:
Added, and it alone produced time ranging from 37s to 67s.

Reply:
reverted, wrong testing data